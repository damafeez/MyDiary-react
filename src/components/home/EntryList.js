import React, { Component } from 'react';
import './EntryList.scss';

export default class AddEntry extends Component {
  state = {
    title: '',
    body: '',
  }

  render() {
    const { title, body } = this.state;
    const { setRef } = this.props;
    return (
      <section ref={setRef} className="entries">
        <div className="today">
          <span id="date">13</span>
          <span id="day">TUESDAY</span>
        </div>
        <ul>
          <li>
            <span className="pointer" />
            <h4 className="title">Funny Title</h4>
            <p className="body">Lorem ipsum, dolor sit amet consectetur adipisicing elit. Blanditiis aliquam corrupti non eius labore facilis dolor doloremque, molestias necessitatibus esse eos dolorum architecto aut fugiat praesentium, delectus ipsum sint veniam.</p>
            <span className="date">JUN<br />2018</span>
          </li>
          <li>
            <span className="pointer" />
            <h4 className="title">Funny Title</h4>
            <p className="body">Lorem ipsum, dolor sit amet consectetur adipisicing elit. Blanditiis aliquam corrupti non eius labore facilis dolor doloremque, molestias necessitatibus esse eos dolorum architecto aut fugiat praesentium, delectus ipsum sint veniam.</p>
            <span className="date">JUN<br />2018</span>
          </li>
        </ul>
      </section>
    );
  }
}
