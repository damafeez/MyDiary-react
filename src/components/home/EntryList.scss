
.entries {
  flex: 0 0 25rem;
  min-width: 30rem;
  min-height: 40rem;
  background: #fcfafa;
  margin-right: 0;
  margin-bottom: 5rem;
  box-shadow: 25px 0px 162px -68px rgba(0, 0, 0, 1);
  border-radius: 4px;
  .today {
    height: 15rem;
    border-radius: 4px 4px 0 0;
    background: linear-gradient(rgba(0, 0, 0, .3), rgba(0, 0, 0, .3)), url(../../assets/bg-4.jpg);
    background-size: 100% auto;
    color: white;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    #date {
      font-size: 4.5rem;
      text-transform: uppercase;
    }
    #day {
      font-size: 1.5rem;
    }
  }
  ul {
    list-style: none;
    padding: 2rem 1rem;
    max-height: 25rem;
    overflow: auto;
    li {
      display: grid;
      grid-template-areas: "pointer title date"
                           "pointer body date";
      grid-template-columns: 2rem 1fr 4rem;
      grid-gap: .5rem .7rem;
      align-items: center;
      cursor: pointer;
      &:not(:last-child) {
        margin-bottom: 2rem;
      }
      &.active .pointer{
        background: var(--outrageous-orange);
      }
      .pointer{
        grid-area: pointer;
        width: .9rem;
        height: .9rem;
        border-radius: 50%;
        box-shadow: 3px 3px 16px rgba(0, 0, 0, .2);
        background: #999895;
      }
      .title{
        grid-area: title;
        font-size: 1.2rem;
        text-overflow: ellipsis;
        overflow: hidden;
        white-space: nowrap;
      }
      .body{
        grid-area: body;
        font-size: 1.3rem;
        color: grey;
        text-overflow: ellipsis;
        overflow: hidden;
        white-space: nowrap;
      }
      .date{
        grid-area: date;
        font-size: 1.2rem;
        color: #43a047;
        text-align: center;
        text-transform: capitalize;
      }
    }
  }
  @media screen and (min-width: 550px) {
    min-width: 25rem;
    margin-bottom: 0;
    margin-right: 5rem;
  }
  @media screen and (min-width: 700px) {
    margin-right: 10rem;
  }
  @media (min-width: 1100px) {
    margin-right: 15rem;
  }
}