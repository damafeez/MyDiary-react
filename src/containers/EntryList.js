import React, { Component } from 'react';
import { connect } from 'react-redux';
import EntryListUI from '../components/home/EntryList';
import scrollAction from '../actions/scroll';
import { getEntries as getEntriesAction, setCurrentEntry } from '../actions/entries';

export class EntryList extends Component {
  componentDidMount = () => {
    const { getEntries } = this.props;
    getEntries();
  }

  render() {
    const {
      scroll,
      getEntriesLoading,
      getEntriesError,
      entries,
      setEntry,
      currentEntry,
    } = this.props;
    return (
      <EntryListUI
        loading={getEntriesLoading}
        error={getEntriesError}
        entries={entries}
        scroll={scroll}
        handleClick={setEntry}
        currentEntry={currentEntry}
      />
    );
  }
}
const mapStateToProps = (state) => ({
  getEntriesLoading: state.entries.getEntriesLoading,
  getEntriesError: state.entries.getEntriesError,
  entries: state.entries.entries,
  currentEntry: state.entries.currentEntry,
});

export default connect(mapStateToProps, {
  scroll: scrollAction,
  getEntries: getEntriesAction,
  setEntry: setCurrentEntry,
})(EntryList);
