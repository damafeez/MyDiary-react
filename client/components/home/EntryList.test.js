import React from 'react';
import EntryList from './EntryList';

const setup = (overrideProps) => {
  const props = {
    updateMode: false,
    currentIndex: 0,
    scroll: jest.fn(),
    handleClick: jest.fn(),
    entries: [{ id: 1, title: 'this title', body: 'this body' }],
  };
  const wrapper = shallow(<EntryList {...props} {...overrideProps} />);
  return { props, wrapper };
};

it('matches snapshot', () => {
  const { wrapper, props } = setup();
  wrapper.find('li').simulate('click');
  expect(props.handleClick).toHaveBeenCalled();
  expect(props.scroll).toHaveBeenCalledWith(false);
  expect(wrapper).toMatchSnapshot();
});
it('matches snapshot :::currentIndex:1', () => {
  const { wrapper } = setup({ currentIndex: 1 });
  expect(wrapper).toMatchSnapshot();
});
