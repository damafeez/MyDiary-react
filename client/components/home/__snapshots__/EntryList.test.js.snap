// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot :::currentIndex:1 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    currentIndex={1}
    entries={
      Array [
        Object {
          "body": "this body",
          "id": 1,
          "title": "this title",
        },
      ]
    }
    handleClick={[MockFunction]}
    scroll={[MockFunction]}
    updateMode={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="today"
        >
          <span
            id="date"
          >
            6
          </span>
          <span
            id="day"
          >
            THURSDAY
          </span>
        </div>,
        <ul>
          <li
            className=""
            onClick={[Function]}
          >
            <span
              className="pointer"
            />
            <h4
              className="title"
            >
              this title
            </h4>
            <p
              className="body"
            >
              this body
            </p>
            <span
              className="date"
            >
              DEC
              <br />
              6
            </span>
          </li>
        </ul>,
      ],
      "className": "entries",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              id="date"
            >
              6
            </span>,
            <span
              id="day"
            >
              THURSDAY
            </span>,
          ],
          "className": "today",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 6,
              "id": "date",
            },
            "ref": null,
            "rendered": 6,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "THURSDAY",
              "id": "day",
            },
            "ref": null,
            "rendered": "THURSDAY",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className=""
              onClick={[Function]}
            >
              <span
                className="pointer"
              />
              <h4
                className="title"
              >
                this title
              </h4>
              <p
                className="body"
              >
                this body
              </p>
              <span
                className="date"
              >
                DEC
                <br />
                6
              </span>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="pointer"
                />,
                <h4
                  className="title"
                >
                  this title
                </h4>,
                <p
                  className="body"
                >
                  this body
                </p>,
                <span
                  className="date"
                >
                  DEC
                  <br />
                  6
                </span>,
              ],
              "className": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "pointer",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "this title",
                  "className": "title",
                },
                "ref": null,
                "rendered": "this title",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "this body",
                  "className": "body",
                },
                "ref": null,
                "rendered": "this body",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "DEC",
                    <br />,
                    6,
                  ],
                  "className": "date",
                },
                "ref": null,
                "rendered": Array [
                  "DEC",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  6,
                ],
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="today"
          >
            <span
              id="date"
            >
              6
            </span>
            <span
              id="day"
            >
              THURSDAY
            </span>
          </div>,
          <ul>
            <li
              className=""
              onClick={[Function]}
            >
              <span
                className="pointer"
              />
              <h4
                className="title"
              >
                this title
              </h4>
              <p
                className="body"
              >
                this body
              </p>
              <span
                className="date"
              >
                DEC
                <br />
                6
              </span>
            </li>
          </ul>,
        ],
        "className": "entries",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                id="date"
              >
                6
              </span>,
              <span
                id="day"
              >
                THURSDAY
              </span>,
            ],
            "className": "today",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 6,
                "id": "date",
              },
              "ref": null,
              "rendered": 6,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "THURSDAY",
                "id": "day",
              },
              "ref": null,
              "rendered": "THURSDAY",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className=""
                onClick={[Function]}
              >
                <span
                  className="pointer"
                />
                <h4
                  className="title"
                >
                  this title
                </h4>
                <p
                  className="body"
                >
                  this body
                </p>
                <span
                  className="date"
                >
                  DEC
                  <br />
                  6
                </span>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="pointer"
                  />,
                  <h4
                    className="title"
                  >
                    this title
                  </h4>,
                  <p
                    className="body"
                  >
                    this body
                  </p>,
                  <span
                    className="date"
                  >
                    DEC
                    <br />
                    6
                  </span>,
                ],
                "className": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "pointer",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "this title",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": "this title",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "this body",
                    "className": "body",
                  },
                  "ref": null,
                  "rendered": "this body",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "DEC",
                      <br />,
                      6,
                    ],
                    "className": "date",
                  },
                  "ref": null,
                  "rendered": Array [
                    "DEC",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    6,
                  ],
                  "type": "span",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    currentIndex={0}
    entries={
      Array [
        Object {
          "body": "this body",
          "id": 1,
          "title": "this title",
        },
      ]
    }
    handleClick={
      [MockFunction] {
        "calls": Array [
          Array [
            0,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    scroll={
      [MockFunction] {
        "calls": Array [
          Array [
            false,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    updateMode={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="today"
        >
          <span
            id="date"
          >
            6
          </span>
          <span
            id="day"
          >
            THURSDAY
          </span>
        </div>,
        <ul>
          <li
            className="active"
            onClick={[Function]}
          >
            <span
              className="pointer"
            />
            <h4
              className="title"
            >
              this title
            </h4>
            <p
              className="body"
            >
              this body
            </p>
            <span
              className="date"
            >
              DEC
              <br />
              6
            </span>
          </li>
        </ul>,
      ],
      "className": "entries",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              id="date"
            >
              6
            </span>,
            <span
              id="day"
            >
              THURSDAY
            </span>,
          ],
          "className": "today",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 6,
              "id": "date",
            },
            "ref": null,
            "rendered": 6,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "THURSDAY",
              "id": "day",
            },
            "ref": null,
            "rendered": "THURSDAY",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="active"
              onClick={[Function]}
            >
              <span
                className="pointer"
              />
              <h4
                className="title"
              >
                this title
              </h4>
              <p
                className="body"
              >
                this body
              </p>
              <span
                className="date"
              >
                DEC
                <br />
                6
              </span>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="pointer"
                />,
                <h4
                  className="title"
                >
                  this title
                </h4>,
                <p
                  className="body"
                >
                  this body
                </p>,
                <span
                  className="date"
                >
                  DEC
                  <br />
                  6
                </span>,
              ],
              "className": "active",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "pointer",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "this title",
                  "className": "title",
                },
                "ref": null,
                "rendered": "this title",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "this body",
                  "className": "body",
                },
                "ref": null,
                "rendered": "this body",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "DEC",
                    <br />,
                    6,
                  ],
                  "className": "date",
                },
                "ref": null,
                "rendered": Array [
                  "DEC",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  6,
                ],
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="today"
          >
            <span
              id="date"
            >
              6
            </span>
            <span
              id="day"
            >
              THURSDAY
            </span>
          </div>,
          <ul>
            <li
              className="active"
              onClick={[Function]}
            >
              <span
                className="pointer"
              />
              <h4
                className="title"
              >
                this title
              </h4>
              <p
                className="body"
              >
                this body
              </p>
              <span
                className="date"
              >
                DEC
                <br />
                6
              </span>
            </li>
          </ul>,
        ],
        "className": "entries",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                id="date"
              >
                6
              </span>,
              <span
                id="day"
              >
                THURSDAY
              </span>,
            ],
            "className": "today",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 6,
                "id": "date",
              },
              "ref": null,
              "rendered": 6,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "THURSDAY",
                "id": "day",
              },
              "ref": null,
              "rendered": "THURSDAY",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="active"
                onClick={[Function]}
              >
                <span
                  className="pointer"
                />
                <h4
                  className="title"
                >
                  this title
                </h4>
                <p
                  className="body"
                >
                  this body
                </p>
                <span
                  className="date"
                >
                  DEC
                  <br />
                  6
                </span>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="pointer"
                  />,
                  <h4
                    className="title"
                  >
                    this title
                  </h4>,
                  <p
                    className="body"
                  >
                    this body
                  </p>,
                  <span
                    className="date"
                  >
                    DEC
                    <br />
                    6
                  </span>,
                ],
                "className": "active",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "pointer",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "this title",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": "this title",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "this body",
                    "className": "body",
                  },
                  "ref": null,
                  "rendered": "this body",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "DEC",
                      <br />,
                      6,
                    ],
                    "className": "date",
                  },
                  "ref": null,
                  "rendered": Array [
                    "DEC",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    6,
                  ],
                  "type": "span",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
